12-http/example3/main.go

```diff
@@ -39,7 +68,9 @@ func usersHandler(w http.ResponseWriter, r *http.Request) {
 	switch r.Method {
 	case "GET", "HEAD":
 		// List the users
+		mu.RLock()
```

Comment:
http://www.makelinux.net/books/lkd2/ch09lev1sec2 explains some of the "don't lock code" reasoning in the "What do I lock?" section.